src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/datasource/OffenderRisksDataSource.kt

```diff
@@ -195,11 +197,104 @@ class CommunityApiOffenderRisksDataSource(
 }
 
 @Component
-class ApDeliusContextApiOffenderRisksDataSource : OffenderRisksDataSource {
+class ApDeliusContextApiOffenderRisksDataSource(
+  private val apDeliusContextApiClient: ApDeliusContextApiClient,
+  private val apOASysContextApiClient: ApOASysContextApiClient,
+  private val hmppsTierApiClient: HMPPSTierApiClient,
+) : OffenderRisksDataSource {
+
+  private val ignoredRegisterTypesForFlags = listOf("RVHR", "RHRH", "RMRH", "RLRH", "MAPP")
+
   override val name: OffenderRisksDataSourceName
     get() = OffenderRisksDataSourceName.AP_DELIUS_CONTEXT_API
 
   override fun getPersonRisks(crn: String): PersonRisks {
-    throw NotImplementedError("Getting risks for individual offenders from the AP Delius Context API is not currently supported")
+    val caseDetailResponse = apDeliusContextApiClient.getCaseDetail(crn)
+
+    return PersonRisks(
+      roshRisks = getRoshRisksEnvelope(crn),
+      mappa = caseDetailResponse.toMappa(),
+      tier = getRiskTierEnvelope(crn),
+      flags = caseDetailResponse.toRiskFlags(),
+    )
+  }
+
+  private fun getRoshRisksEnvelope(crn: String): RiskWithStatus<RoshRisks> {
+    return when (val roshRisksResponse = apOASysContextApiClient.getRoshRatings(crn)) {
+      is ClientResult.Success -> {
+        val summary = roshRisksResponse.body.rosh
+
+        when (summary.anyRisksAreNull()) {
+          true -> RiskWithStatus(status = RiskStatus.NotFound)
+          false -> RiskWithStatus(
+            value = RoshRisks(
+              overallRisk = summary.determineOverallRiskLevel().text,
+              riskToChildren = summary.riskChildrenCommunity!!.text,
+              riskToPublic = summary.riskPublicCommunity!!.text,
+              riskToKnownAdult = summary.riskKnownAdultCommunity!!.text,
+              riskToStaff = summary.riskStaffCommunity!!.text,
```

Comment:
These bangs make me nervous - can we guarantee these fields will be available?

Reply:
The `summary.anyRisksAreNull()` method would return `true` if any of these fields were null, so this branch would never be executed.